components/table/table-col-sort-button.js

```diff
@@ -81,19 +92,54 @@ export class TableColSortButton extends FocusMixin(LitElement) {
 
 	constructor() {
 		super();
-		this.nosort = false;
 		this.desc = false;
+		this.nosort = false;
+		this.sourceType = 'unknown';
+
+		this._describedById = getUniqueId();
 	}
 
 	static get focusElementSelector() {
 		return 'button';
 	}
 
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		if (!this.sourceType || this.sourceType === 'unknown') {
+			console.warn('d2l-table-col-sort-button: source-type attribute does not have a value specified.');
+		}
+	}
+
 	render() {
+		const buttonDescription = this.nosort ? this.localize('components.table-col-sort-button.addSortOrder') : this.localize('components.table-col-sort-button.changeSortOrder');
+		const buttonTitle = this.nosort
+			? undefined
+			: this.localize('components.table-col-sort-button.title', {
+				sourceType: this.sourceType,
+				direction: this.desc ? 'desc' : undefined
+			});
+
 		const iconView = !this.nosort ?
 			html`<d2l-icon icon="${this.desc ? 'tier1:arrow-toggle-down' : 'tier1:arrow-toggle-up'}"></d2l-icon>` :
 			null;
-		return html`<button type="button"><slot></slot>${iconView}</button>`;
+
+		return html`<button
+				aria-describedby="${this._describedById}"
+				title="${ifDefined(buttonTitle)}"
+				type="button">
+				<slot></slot>${iconView}
+			</button><span id="${this._describedById}" hidden>${buttonDescription}</span>`;
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		if (!changedProperties.has('sourceType')) return;
+
+		if (!this.sourceType || this.sourceType === 'unknown') {
+			console.warn('d2l-table-col-sort-button: source-type attribute does not have a value specified.');
```

Comment:
I'd be tempted to remove this and the warning in `firstUpdated` as well. It's going to be spammy, clients are going to see it, and since we have a decent (good enough for the past several years) fallback, maybe it's OK to just leave them as-is and treat setting this as a strong recommendation but not warning-worthy?

Reply:
I'm also good with that